extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/interceptor/AbstractMethodValidationInterceptor.java

```diff
@@ -37,9 +40,12 @@ public abstract class AbstractMethodValidationInterceptor implements Serializabl
      * field here. Upon passivation not the validator itself will be serialized, but
      * the proxy injected here, which in turn is serializable.
      * </p>
+     * {@link Instance} is used to make the resolution dynamic and working at runtime, this delays {@link Validator}
+     * injection enough to enable it to work for beans that observe {@code @Initialized(ApplicationScoped.class)} event.
      */
     @Inject
-    Validator validator;
+    @WithCaching
```

Comment:
If I move to singleton beans for ValidatorFactory and Validator, I can remove this, right? I'll check later today if something complains in the tree.

Reply:
That's correct (and even if you kept it here, it would simply have zero effect, it only affects resolution of dependent beans).